### DISCUSSION TITLE
"isDirty" formState remains "false" after removing item from field array

### DISCUSSION THREAD

**Discussion by PlaygroundA** [react-hook-form/react-hook-form#3360] [Q&A] [answered]
Hi everyone! I have a question about "onDirty" formState. 

I have a form, consisting of field array and several controls. And the problem is: 
    - When I change one of the input fields, "is Dirty" state becomes true - which is Ok. Then, when you append new array item, "isDirty " remains the same  -  which is also ok, but when you remove the array item you just added, "isDirty" becomes false, even though some of the controls have changed.

Here is CSB example. 
[https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dq1fh?file=/src/index.js](https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dq1fh?file=/src/index.js)
To reproduce what I am saying, please do the following:
1. Open console
2. Change 1st input value (default value = 'hello'). In console you will see "IS DIRTY  - true"
3. Click Append to add new array item - is Dirty remains the same
3 Remove new array item. In console you will see - "IS DIRTY - false"

What is the correct behavior? 
I think if you change some field, then the formState.isDirty should be true, despite adding/removing field array items.

What I'm doing wrong?

**Comment by bluebill1049** [comment, marked as answer]
can you test with the the latest version?

**Reply by PlaygroundA** [reply]
I'm using latest version in my project. 

But, just in case, i've updated CSB example with 6.10.1 version. The same
[https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dq1fh?file=/src/fieldArray.js](https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-dq1fh?file=/src/fieldArray.js)

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-forked-udeui?file=/src/fieldArray.js
```
append({
  name: ''
})
```

**Reply by PlaygroundA** [reply]
Sorry ( It didn't work. 

I took these steps  (in your example):
1. Changed 'hello' => isDirty = true
2. Append new item => isDirty = true
3. Remove just added item => isDirty = false  (but 'hello' has been changed)

**Reply by bluebill1049** [reply]
oh... that's a bug.

**Reply by bluebill1049** [reply]
you want to report a bug?

**Reply by PlaygroundA** [reply]
Yes, I'll create a bug report a bit later  (but today)

Thank you for answering )

**Reply by bluebill1049** [reply]
thank you mate, i have fixed it last night, will release the minor version this weekend.

**Comment by denisesenguel** [comment]
Hey! I'm facing a very similar issue. When removing an item from an array, I would expect the whole array to be marked as dirty, which is not the case. To reproduce: remove second item here: https://codesandbox.io/s/dlpcfc?file=/src/index.js:145-168

**Comment by w4-utae** [comment]
Is there any update about this issue?
